src/data/Nurse.h

```diff
@@ -76,6 +77,7 @@ class Contract{
    friend std::ostream& operator<< (std::ostream& outs, Contract obj) {return outs << obj.toString();}
 };
 
+typedef std::shared_ptr<const Contract> PConstContract;
```

Comment:
Move declaration to Scenario.h? This way, you will be able to use this type everywhere, e.g., lines 128 and 183 of Scenario.h.

Reply:
Done on d65a2e4f0720a5cf4f5d2dde0af931284ab91afc